dspace/config/dspace.cfg

```diff
@@ -107,77 +107,37 @@ mail.allowed.referrers = ${dspace.hostname}
 # This is especially useful for development and test environments where production data is used when testing functionality.
 #mail.server.disabled = false
 
-##### File Storage ######
+##### Asset Storage (bitstreams / files) ######
+# Prior to DSpace 6, assetstore was local filesystem, using assetstore.dir style configs
+# Starting with DSpace 6, different storage implementations are available.
+# - DSBitstore - Store on local filesystem in directory scatter pattern (i.e. 10/11/12/101112xxxxx)
+# - S3Bitstore - Store in AWS S3 cloud storage
 
+# DEPRECATED: Legacy assetstore configuration, stop using it
 # Asset (bitstream) store number 0 (zero)
-assetstore.dir = ${dspace.dir}/assetstore
-
-# Specify extra asset stores like this, counting from 1 upwards:
+# assetstore.dir = ${dspace.dir}/assetstore
 # assetstore.dir.1 = /second/assetstore
 # assetstore.dir.2 = /third/assetstore
 
+## DSpace 6+ assetstore config
+# Configure the bitstore implementations
+## bitstore.<short-name>.class=<implementation-class-canonical-name>
+bitstore.ds.class=org.dspace.storage.bitstore.DSBitStore
+bitstore.s3.class=org.dspace.storage.bitstore.S3BitStore
```

Comment:
We do already have similar solutions internally. An ideal configuration would look something like the following.

```
    <bean name="org.dspace.storage.bitstore.BitstreamStorageService" class="org.dspace.storage.bitstore.BitstreamStorageServiceImpl">
        <property name="defaultIncomingBitStore" value="0"/>
        <property name="bitStores">
            <map>
                <entry key="0" value-ref="localStore"/>
            </map>
        </property>
    </bean>

    <bean name="localStore" class="org.dspace.storage.bitstore.storeimpl.DSBitStore" scope="singleton">
        <property name="baseDir" value="${dspace.dir}/assetstore"/>
    </bean>
```